packages/common/src/solana/TransactionV2.ts

```diff
@@ -0,0 +1,29 @@
+import { Transaction, VersionedTransaction } from "@solana/web3.js";
+import { ethers } from "ethers";
+const { base58: bs58 } = ethers.utils;
+
+export class TransactionV2 {
+  static from(serializedTx: string): Transaction | VersionedTransaction {
+    try {
+      const transaction = Transaction.from(bs58.decode(serializedTx));
+      return transaction;
+    } catch (e) {
+      return VersionedTransaction.deserialize(bs58.decode(serializedTx));
+    }
+  }
+  static isVersioned(tx: Transaction | VersionedTransaction) {
+    if ("version" in tx) {
+      return true;
+    }
+    return false;
+  }
```

Comment:
```suggestion
  static isVersioned(tx: Transaction | VersionedTransaction): tx is VersionedTransaction {
    return "version" in tx;
  }
```

This way you get type narrowing when you call this.